### PR TITLE
build logs view + data in psql

### LINKED ISSUE #1: Store and provide logs in packit service
get logs from copr realtime and provide them realtime in a sensible way (show what matters to users, not internal infra stuff people don't understand)

### LINKED ISSUE #264: improve redis server handling in tests
openshift requre tests uses running redis server, it is hardcoded in bash script running tests with ``sleep 5`` to ensure that server is running.
https://github.com/packit-service/packit-service/blob/master/files/run_tests.sh#L12

Would be better to replace it by some opeshift magic, e.g. run redis server is dedicated pod, and tests will use it.

### PR DESCRIPTION
**Pull request by TomasTomecek**
replaces #257 #297

Fixes #1
Fixes #264

 * [x] tests pass in CI                                                                                                                                                       
   * [x] utilize pg and redis deployed w/ requre tests                                                                                                                        
 * [x] have a succ build on stg w/ good logs
 * move PR comments to logs (new PR)
 * [x] SRPM check has logs

### CODE REVIEW

File: packit_service/models.py [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+
+
+# https://github.com/python/mypy/issues/2477#issuecomment-313984522 ^_^
+if TYPE_CHECKING:
+    Base = object
+else:
+    Base = declarative_base()
+
+
+class GitHubProject(Base):
```
Comment (TomasTomecek) [inline packit_service/models.py]: This is the PSQL schema.

**Comment by TomasTomecek** [comment]
> Build succeeded.
> 
>     * [packit-service-pre-commit ](https://softwarefactory-project.io/zuul/t/local/build/c6814080d8b94080a6cbb650b87f3f0d) : SUCCESS in 3m 14s
> 
>     * [packit-service-tests ](https://softwarefactory-project.io/zuul/t/local/build/dcd2c446e13e4ca981b8d794173db51c) : SUCCESS in 13m 43s
> 
>     * [packit-service-tests-requre ](https://softwarefactory-project.io/zuul/t/local/build/f381a7b3e9fb4656bfcedb52fcfd5ad2) : SUCCESS in 15m 35s

WHAT?! that's not possible

File: files/install-deps-worker.yaml [hunk 2]
```diff
@@ -12,6 +12,8 @@
           - python3-kubernetes # for sandcastle
           - python3-fedora # to access FAS
           - python3-requests
+          - python3-flask # sorry jirko
+          - python3-flask-restplus
```
Comment (jpopelka) [inline files/install-deps-worker.yaml]: `python3-flask-restplus` is enough because it requires `python3-flask`.

File: alembic/env.py [hunk 3]
```diff
@@ -29,7 +29,8 @@
 def get_pg_url() -> str:
     url = (
         f"postgres+psycopg2://{os.getenv('POSTGRESQL_USER')}"
-        f":{os.getenv('POSTGRESQL_PASSWORD')}@postgres:5432/{os.getenv('POSTGRESQL_DATABASE')}"
+        f":{os.getenv('POSTGRESQL_PASSWORD')}@{os.getenv('POSTGRESQL_HOST', 'postgres')}"
+        f":5432/{os.getenv('POSTGRESQL_DATABASE')}"
```
Comment (csomh) [inline alembic/env.py]: How about making the port also configurable, with a default of "5432"?

File: packit_service/service/urls.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+# SOFTWARE.
+from typing import Optional
+
+from flask import url_for
+
+from packit_service.service.app import application
+from packit_service.service.events import CoprBuildEvent
+
+
+def get_p_s_logs_url(
```
Comment (csomh) [inline packit_service/service/urls.py]: I find this name a somewhat criptic and hard to remember. Would `get_build_log_url` or `get_log_url` work?

File: packit_service/models.py [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+    build_id = Column(String, index=True)  # copr build id
+    pr_id = Column(Integer, ForeignKey("pull_requests.id"))
+    pr = relationship("PullRequest", back_populates="copr_builds")
+    logs = Column(Text)
+    commit_sha = Column(String)
+    status = Column(String)
+    target = Column(String)
+    web_url = Column(String)  # URL to web ui
+    build_logs_url = Column(String)  # url to copr build logs
+    logs_url = Column(String)
```
Comment (csomh) [inline packit_service/models.py]: Might be good to have a more explicit documentation for these fields, for example to specify which service these URLs are pointing to: copr or Packit Service.

File: alembic/env.py (hunk 3 above)
Comment (TomasTomecek) [reply alembic/env.py]: sure :D

Comment (TomasTomecek) [reply alembic/env.py]: actually, how about we imported this method from packit_service :O

File: packit_service/service/urls.py (hunk 4 above)
Comment (TomasTomecek) [reply packit_service/service/urls.py]: get_log_url lgtm

**Comment by TomasTomecek** [comment]
thanks for the comments, pushed changes to address them, also realised the schema is not applied, hence the sentry issue https://sentry.io/organizations/red-hat-0p/issues/1518475677/

built && pushed new stg image, let's see tomorrow

**Comment by csomh** [comment]
Tests fail because

```
Unable to find '../secrets/dev/dashboard-privkey.pem' in expected
```

@sakalosj I think the tests should be updated to set up `dashboard-privkey.pem` after the recent deployment changes.

**Comment by TomasTomecek** [comment]
> Tests fail because
> 
> ```
> Unable to find '../secrets/dev/dashboard-privkey.pem' in expected
> ```
> 
> @sakalosj I think the tests should be updated to set up `dashboard-privkey.pem` after the recent deployment changes.

shoiuld be fixed in https://github.com/packit-service/packit-service/pull/406/commits/2842c46e63a5f9b6091b93ebf239ab8e44fd703f

**Comment by sakalosj** [comment]
> > Tests fail because
> > ```
> > Unable to find '../secrets/dev/dashboard-privkey.pem' in expected
> > ```
> > 
> > 
> > @sakalosj I think the tests should be updated to set up `dashboard-privkey.pem` after the recent deployment changes.
> 
> shoiuld be fixed in [2842c46](https://github.com/packit-service/packit-service/commit/2842c46e63a5f9b6091b93ebf239ab8e44fd703f)

thanks - wasnt aware of this

**Comment by TomasTomecek** [comment]
#414 to fix CI

**Comment by TomasTomecek** [comment]
recheck

**Comment by TomasTomecek** [comment]
Finally a successful build, merging https://github.com/packit-service/hello-world/pull/78

**Pull request merged**